src/glue_launcher_lambda/glue_launcher.py

```diff
@@ -0,0 +1,517 @@
+    )
+
+    return detail_dict
+
+
+def generate_ms_epoch_from_timestamp(formatted_timestamp_string, minutes_to_add=0):
+    """Returns the 1970 epoch as a number from the given timestamp.
+
+    Keyword arguments:
+    timestamp_string -- the timestamp as a string formatted to %Y-%m-%dT%H:%M:%S.%f%z
+    minutes_to_add -- if any minutes are to be added to the time, set to greater than 0
+    """
+    timestamp = datetime.strptime(formatted_timestamp_string, "%Y-%m-%dT%H:%M:%S.%f")
+    if minutes_to_add > 0:
+        timestamp = timestamp + timedelta(minutes=minutes_to_add)
+
+    return int((timestamp - datetime(1970, 1, 1)).total_seconds() * 1000.0)
+
+
+def get_batch_client():
+    return boto3.client("batch")
+
+
+def get_athena_client():
+    return boto3.client("athena")
+
+
+def get_glue_client():
+    return boto3.service("glue")
+
+
+def check_running_batch_tasks(job_queue, batch_client):
+    """Check the AWS Batch job queue, return count of tasks in each status"""
+
+    operational_tasks = 0
+
+    for status in OPERATIONAL_JOB_STATUSES:
+        response = batch_client.list_jobs(
+            jobQueue=job_queue,
+            jobStatus=status,
+        )
+        operational_tasks += len(response["jobSummaryList"])
+    return operational_tasks
+
+
+def fetch_table_creation_sql_files(file_path, args=None):
+    with open(os.path.join(file_path, "create-parquet-table.sql"), "r") as f:
+        base_create_parquet_query = f.read()
+
+    with open(
+        os.path.join(file_path, "create-missing-import-table.sql"),
+        "r",
+    ) as f:
+        base_create_missing_import_query = f.read()
+
+    with open(
+        os.path.join(file_path, "create-missing-export-table.sql"),
+        "r",
+    ) as f:
+        base_create_missing_export_query = f.read()
+
+    with open(os.path.join(file_path, "create-count-table.sql"), "r") as f:
+        base_create_count_query = f.read()
+
+    tables = [
+        [
+            args.missing_imports_table_name,
+            base_create_missing_import_query,
+            args.manifest_s3_input_parquet_location_missing_import,
+        ],
+        [
+            args.missing_exports_table_name,
+            base_create_missing_export_query,
+            args.manifest_s3_input_parquet_location_missing_export,
+        ],
+        [
+            args.counts_table_name,
+            base_create_count_query,
+            args.manifest_s3_input_parquet_location_counts,
+        ],
+        [
+            args.mismatched_timestamps_table_name,
+            base_create_parquet_query,
+            args.manifest_s3_input_parquet_location_mismatched_timestamps,
+        ],
+    ]
+
+    return tables
+
+
+def fetch_table_drop_sql_file(file_path, args=None):
+    with open(os.path.join(file_path, "drop-table.sql"), "r") as f:
+        base_drop_query = f.read()
+        return base_drop_query
+
+
+def poll_athena_query_status(id):
```

Comment:
I can't see a test for this method that ensures the client itself has the right calls

Reply:
Also we should pass the athena client in here